aries_vcx/tests/test_integration.rs

```diff
@@ -1001,38 +1014,104 @@ mod tests {
-        consumer_to_institution.update_state().unwrap();
+        let oob_receiver = OutOfBand::create_from_a2a_msg(&oob_msg).unwrap();
+        let conn = oob_receiver.connection_exists(vec![]).unwrap();
+        assert!(!conn.is_some());
+        let mut conn_receiver = oob_receiver.build_connection(true).unwrap();
+        conn_receiver.connect().unwrap();
+        conn_receiver.update_state().unwrap();
+        assert_eq!(ConnectionState::Invitee(InviteeState::Requested), conn_receiver.get_state());
+        assert_eq!(oob_sender.id.0, oob_receiver.id.0);
 
         institution.activate().unwrap();
         thread::sleep(Duration::from_millis(500));
-        let mut conn_requests = institution.agent.download_connection_requests(None).unwrap();
-        assert_eq!(conn_requests.len(), 1);
-        let mut institution_to_consumer = Connection::create_with_connection_request(conn_requests.pop().unwrap(), &institution.agent).unwrap();
-        assert_eq!(ConnectionState::Inviter(InviterState::Requested), institution_to_consumer.get_state());
-        institution_to_consumer.update_state().unwrap();
-        assert_eq!(ConnectionState::Inviter(InviterState::Responded), institution_to_consumer.get_state());
+        let mut requests = institution.agent.download_connection_requests(None).unwrap();
+        assert_eq!(requests.len(), 1);
+        let mut conn_sender = Connection::create_with_connection_request(requests.pop().unwrap(), &institution.agent).unwrap();
+        assert_eq!(ConnectionState::Inviter(InviterState::Requested), conn_sender.get_state());
+        conn_sender.update_state().unwrap();
+        assert_eq!(ConnectionState::Inviter(InviterState::Responded), conn_sender.get_state());
 
         consumer.activate().unwrap();
-        consumer_to_institution.update_state().unwrap();
-        assert_eq!(ConnectionState::Invitee(InviteeState::Completed), consumer_to_institution.get_state());
+        thread::sleep(Duration::from_millis(500));
+        conn_receiver.update_state().unwrap();
+        assert_eq!(ConnectionState::Invitee(InviteeState::Completed), conn_receiver.get_state());
 
         institution.activate().unwrap();
         thread::sleep(Duration::from_millis(500));
-        institution_to_consumer.update_state().unwrap();
-        assert_eq!(ConnectionState::Inviter(InviterState::Completed), institution_to_consumer.get_state());
+        conn_sender.update_state().unwrap();
+        assert_eq!(ConnectionState::Inviter(InviterState::Completed), conn_sender.get_state());
 
-        institution_to_consumer.send_generic_message("Hello Alice, Faber here").unwrap();
+        let conn = oob_receiver.connection_exists(vec![&conn_receiver]).unwrap();
+        assert!(conn.is_some());
+
+        let a2a_msg = oob_receiver.extract_a2a_message().unwrap().unwrap();
+        assert!(matches!(a2a_msg, A2AMessage::PresentationRequest(..)));
+        if let A2AMessage::PresentationRequest(request_receiver) = a2a_msg {
```

Comment:
I suggest only using BasicMessage for this test, not a big deal but:
- no need to create schema and cred def
- it's clearer this test is not testing oob+proofs, as it might appear at quick glance

Reply:
True (I did that just to generate a reasonable presentation request)